### DISCUSSION TITLE
Putting code blocks in lists?

### DISCUSSION THREAD

**Discussion by matthijskooijman** [outline/outline#8129] [General] [answered]
Does outline somehow supports putting a code block inside a list/enumeration? With the editor, I only seem to be able to make code blocks at the top level (with ```` ``` ```` or by selecting the block).

In (at least some flavors of) Markdown, this is possible by indenting the code block to the same level as the list. E.g. this markdown:

````
1. This is a list
   ```
   with a code block
   of two lines
   ```
2. and a second item
````

Produces this output:

1. This is a list
   ```
   with a code block
   of two lines
   ```
2. and a second item

Which is what I would like to do in Outline as well. Outline does seem to support the structure, since when I paste the above markdown source into Outline, I get what I want:

![image](https://github.com/user-attachments/assets/8befb4b2-05bf-4979-ac60-dee25d4c2209)

But when I type it manually (using shift-enter to continue the first list item), it does not recognize this at all:

![image](https://github.com/user-attachments/assets/bdf1318b-b6fa-4ccc-b41a-5b976055c422)

The closest thing I could manage is to use single backticks to get inline code formatting, but that does not look so good (only the text gets a background instead of getting a rectangle), does not support other formatting like mermaid diagrams or so, and does not have the easy copy-to-clipboard button. E.g.:

![image](https://github.com/user-attachments/assets/aec562bf-c188-4ee8-909f-28473616bf90)

Is there a way that I'm missing to easily get this without resorting to an external text editor to type markdown first?

This was tested with the latest Outline, 0.81.1.

**Comment by tommoor** [comment, marked as answer]
It should be supported, I'm not sure what changes will be required in the depths of the editor to have these commands work after a shift-enter (hardbreak) though. I notice even doing `/code` in a list does not work correctly, definitely a little buggy.

File as an issue?

**Reply by matthijskooijman** [reply]
Thanks for confirming. I've filed an issue here: https://github.com/outline/outline/issues/8216

**Comment by Havoc45** [comment]
I literally having this issue right now! Creating docs is difficult as the flow is somewhat broken with caption or description is separated from the code block. And you can't drag and drop a code block which sucks. I'm keeping my eye on this. Thanks for reporting this first.

I'm on the licensed edition btw.

**Reply by tommoor** [reply]
Sidenote: You can drag a code block by selecting it with `cmd+click` and then dragging

**Reply by Havoc45** [reply]
Hi @tommoor. TIL, thanks for highlighting this. This makes my life a bit easier to writing all the docs I'm creating on right now 🥲
